### PR TITLE
[API] Ensure string dictionary keys casing gets respected [WIP]

### PR DESCRIPTION
**Pull request by fassadlr**
Don't merge, awaiting test results!

### CODE REVIEW

**Comment by bokobza** [comment]
I wonder what prompted this change.
Can you please enlighten me @fassadlr ?

**Comment by fassadlr** [comment]
@zeptin I'm not yet sure whether the change I did is the right one. If I override the serializer settings then the node complained about `Money` needing a parameter-less constructor for serializing/deserialization.

Perhaps we should not serialize `Money` and just pass/return `decimal`? Also maybe it will be better to just serialize a model with a list of addresses and balances instead of a dictionary? Then we won't have these serialization changes.

@bokobza what brought this change was the fact that dictionary keys gets serialized with a lowercase if its a string. In this case they are STRAT addresses which can't change as they are case sensitive. This caused "duplicates" to be added to the deposits collection on the side chain side.

**Comment by fassadlr** [comment]
OK I have changed the controllers to return a model 👍

File: src/NBitcoin/Money.cs [hunk 1]
```diff
@@ -11,10 +11,10 @@ public static class MoneyExtensions
     {
         public static Money Sum(this IEnumerable<Money> moneys)
         {
-            if(moneys == null)
+            if (moneys == null)
```
Comment (dangershony) [inline src/NBitcoin/Money.cs]: I can't find what has changed in this class `Money`

Comment (zeptin) [reply src/NBitcoin/Money.cs]: @dangershony there was an empty constructor added in an earlier commit, this has been reverted.

**Pull request merged**